helios-services/src/main/java/com/spotify/helios/agent/DefaultSecretVolumeManager.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.agent;
+
+
+import com.spotify.helios.common.SecretVolumeException;
+import com.spotify.helios.common.descriptors.SecretVolumeRequest;
+import com.spotify.helios.common.descriptors.Secrets;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.file.Paths;
+
+import okhttp3.Call;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+
+public class DefaultSecretVolumeManager implements SecretVolumeManager {
+  private static final Logger log = LoggerFactory.getLogger(DefaultSecretVolumeManager.class);
+
+  private static final MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+  private static final String SCHEME = "http";
+
+  private final OkHttpClient client;
+  private final SmallFileReader reader;
+  private final String baseUrl;
+  private final String path;
+
+  public DefaultSecretVolumeManager(final String baseUrl, final String path) {
+    this(new OkHttpClient(), new FsSmallFileReader(), baseUrl, path);
+  }
+
+  public DefaultSecretVolumeManager(final OkHttpClient client,
+                                    final SmallFileReader fileReader,
+                                    final String baseUrl,
+                                    final String path) {
+    this.client = client;
+    this.reader = fileReader;
+    this.baseUrl = baseUrl;
+    this.path = path;
+  }
+
+  public static String urlFromAddr(final InetSocketAddress addr) {
+    return new HttpUrl.Builder()
+        .scheme(SCHEME)
+        .host(addr.getHostString())
+        .port(addr.getPort())
+        .build()
+        .toString();
+  }
+
+  @Override
+  public String create(final String id, final Secrets secretsCfg) throws SecretVolumeException {
+    log.debug("Creating secret volume id {}", id);
+    final SecretVolumeRequest svr = secretVolumeRequest(id, secretsCfg);
+
+    final HttpUrl url = HttpUrl.parse(baseUrl)
```

Comment:
it would make more sense to keep `baseUrl` as the original `HttpUrl`, so that you don't keep re-parsing it on each request.

i.e. you take an InetSocketAddress and turn it into a HttpUrl then turn that into a String, just to turn that back into a HttpUrl here